go/mysql/replication_test.go

```diff
@@ -88,14 +89,50 @@ func TestComBinlogDumpGTID(t *testing.T) {
 		cConn.Close()
 	}()
 
+	t.Run("WriteComBinlogDumpGTIDEmptyGTID", func(t *testing.T) {
+		// Write ComBinlogDumpGTID packet, read it, compare.
+		var flags uint16 = 0x0d0e
+		err := cConn.WriteComBinlogDumpGTID(0x01020304, "moofarm", 0x05060708090a0b0c, flags, []byte{})
+		assert.NoError(t, err)
+		data, err := sConn.ReadPacket()
+		require.NoError(t, err, "sConn.ReadPacket - ComBinlogDumpGTID failed: %v", err)
+		require.NotEmpty(t, data)
+		require.EqualValues(t, data[0], ComBinlogDumpGTID)
+
+		expectedData := []byte{
+			ComBinlogDumpGTID,
+			0x0e, 0x0d, // flags
+			0x04, 0x03, 0x02, 0x01, // server-id
+			0x07, 0x00, 0x00, 0x00, // binlog-filename-len
+			'm', 'o', 'o', 'f', 'a', 'r', 'm', // bilog-filename
+			0x0c, 0x0b, 0x0a, 0x09, 0x08, 0x07, 0x06, 0x05, // binlog-pos
+			0x00, 0x00, 0x00, 0x00, // data-size is zero, no GTID payload
+		}
+		assert.Equal(t, expectedData, data)
+		logFile, logPos, pos, err := sConn.parseComBinlogDumpGTID(data)
+		require.NoError(t, err, "parseComBinlogDumpGTID failed: %v", err)
+		assert.Equal(t, "moofarm", logFile)
+		assert.Equal(t, uint64(0x05060708090a0b0c), logPos)
+		assert.True(t, pos.IsZero())
+	})
+
+	sConn.sequence = 0
+
 	t.Run("WriteComBinlogDumpGTID", func(t *testing.T) {
 		// Write ComBinlogDumpGTID packet, read it, compare.
 		var flags uint16 = 0x0d0e
 		assert.Equal(t, flags, flags|BinlogThroughGTID)
-		err := cConn.WriteComBinlogDumpGTID(0x01020304, "moofarm", 0x05060708090a0b0c, flags, []byte{0xfa, 0xfb})
+		gtidSet, err := replication.ParseMysql56GTIDSet("16b1039f-22b6-11ed-b765-0a43f95f28a3:1-243")
+		require.NoError(t, err)
+		sidBlock := gtidSet.SIDBlock()
+		assert.Len(t, sidBlock, 0x30) // 48 bytes
```

Comment:
Nit, but no reason not to use decimal here (can then remove the comment).

Reply:
done.